gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs/components/api-runtime-logs-quick-filters/api-runtime-logs-quick-filters.component.ts

```diff
@@ -14,78 +14,94 @@
  * limitations under the License.
  */
 
-import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
 import { distinctUntilChanged, takeUntil } from 'rxjs/operators';
 import { isEqual } from 'lodash';
 import { Subject } from 'rxjs';
 import { KeyValue } from '@angular/common';
 
-export type PeriodFilter = {
-  label: string;
-  value: string;
-};
-
-export type LogFilter = {
-  period?: PeriodFilter;
-};
+import { DEFAULT_PERIOD, LogFilter, LogFiltersInitialValues, MultiFilter, PERIODS, SimpleFilter } from './models';
+import { CacheEntry } from './components';
 
 @Component({
   selector: 'api-runtime-logs-quick-filters',
   template: require('./api-runtime-logs-quick-filters.component.html'),
   styles: [require('./api-runtime-logs-quick-filters.component.scss')],
 })
 export class ApiRuntimeLogsQuickFiltersComponent implements OnInit, OnDestroy {
-  private readonly defaultPeriod: PeriodFilter = { label: 'None', value: '0' };
-  private defaultFilter: LogFilter = {
-    period: this.defaultPeriod,
-  };
-
-  public readonly periods: PeriodFilter[] = [
-    this.defaultPeriod,
-    { label: 'Last 5 Minutes', value: '-5m' },
-    { label: 'Last 30 Minutes', value: '-30m' },
-    { label: 'Last 1 Hour', value: '-1h' },
-    { label: 'Last 3 Hours', value: '-3h' },
-    { label: 'Last 6 Hours', value: '-6h' },
-    { label: 'Last 12 Hours', value: '-12h' },
-    { label: 'Last 1 Day', value: '-1d' },
-    { label: 'Last 3 Days', value: '-3d' },
-    { label: 'Last 7 Days', value: '-7d' },
-  ];
-
-  public isFiltering = false;
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
 
-  public quickFiltersForm = new FormGroup({
-    period: new FormControl(this.defaultPeriod),
-  });
+  @Input() initialValues: LogFiltersInitialValues;
+  @Output() quickFilterSelection = new EventEmitter<LogFilter>();
+  defaultFilter: LogFilter = {
+    period: DEFAULT_PERIOD,
+    applications: undefined,
+  };
+  readonly periods = PERIODS;
+  isFiltering = false;
+  quickFiltersForm: FormGroup;
+  currentFilter: LogFilter;
+  applicationsCache: CacheEntry[];
 
-  public currentFilter: LogFilter = {};
+  ngOnInit(): void {
+    this.currentFilter = {
+      ...this.defaultFilter,
+      applications: this.initialValues.applications ?? this.defaultFilter?.applications,
+    };
 
-  @Output()
-  quickFilterSelection = new EventEmitter<LogFilter>();
+    this.quickFiltersForm = new FormGroup({
+      period: new FormControl(DEFAULT_PERIOD),
+      applications: new FormControl(
+        this.initialValues.applications?.map((application) => application.value) ?? this.defaultFilter?.applications,
+      ),
+    });
 
-  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+    this.onPeriodFilterChange();
+    this.onApplicationFilterChange();
```

Comment:
Would it be possible to only handling one subscription on the whole group?

Reply:
Done